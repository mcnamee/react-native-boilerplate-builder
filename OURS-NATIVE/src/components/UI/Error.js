import React from 'react';
import PropTypes from 'prop-types';
import {
  Container, Text, H3, Button, View,
} from 'native-base';
import Spacer from './Spacer';

const Error = ({ title, content, tryAgain }) => (
  <Container style={{ flex: 1 }}>
    <View style={{ alignSelf: 'center' }}>
      <Spacer size={20} />
      <H3 style={{ textAlign: 'center' }}>{title}</H3>
      <Text style={{ textAlign: 'center', marginBottom: 20 }}>{content}</Text>
      {tryAgain && (
        <Button block onPress={tryAgain}>
          <Text>Try Again</Text>
        </Button>
      )}
      <Spacer size={20} />
    </View>
  </Container>
);

Error.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
  tryAgain: PropTypes.func,
};

Error.defaultProps = {
  title: 'Uh oh',
  content: 'An unexpected error came up',
  tryAgain: null,
};

export default Error;
