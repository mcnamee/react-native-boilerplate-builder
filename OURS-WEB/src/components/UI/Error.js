import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'reactstrap';
// import { Link } from 'react-router-dom';

const Error = ({ title, content }) => (
  <Row className="py-5 my-5">
    <Col md={{ size: 4, offset: 4 }}>
      <h2 className="text-center">{title}</h2>
      <p className="text-center">{content}</p>
      {/* <p className="text-center"><Link to="/" className="btn btn-primary">Go Home</Link></p> */}
    </Col>
  </Row>
);

Error.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
};

Error.defaultProps = {
  title: 'Uh oh',
  content: 'An unexpected error came up',
};

export default Error;
