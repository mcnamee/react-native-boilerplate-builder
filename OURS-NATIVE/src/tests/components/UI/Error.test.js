import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';
import Error from '../../../components/UI/Error';

it('<Error /> renders with message', () => {
  const Component = <Error title="hello boy" />;

  // Matches snapshot
  expect(renderer.create(Component).toJSON()).toMatchSnapshot();

  // Has the correct text on the page
  const { getByText } = render(Component);
  expect(getByText('hello boy'));
});

it('<Error /> renders with a button', () => {
  const Component = <Error title="hello boy" tryAgain={() => {}} />;

  // Matches snapshot
  expect(renderer.create(Component).toJSON()).toMatchSnapshot();

  // Has the correct text on the page
  const { getByText } = render(Component);
  expect(getByText('hello boy'));
  expect(getByText('Try Again'));
});
