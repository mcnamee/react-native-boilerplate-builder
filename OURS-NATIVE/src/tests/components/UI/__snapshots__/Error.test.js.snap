// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders with a button 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignSelf": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "height": 20,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 21,
            "lineHeight": 25,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      hello boy
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "marginBottom": 20,
            "textAlign": "center",
          },
        ]
      }
      uppercase={false}
    >
      An unexpected error came up
    </Text>
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#007aff",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 2,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
        }
      }
    >
      <Text
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          }
        }
        uppercase={false}
      >
        Try Again
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "height": 20,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`<Error /> renders with message 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignSelf": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "height": 20,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontSize": 21,
            "lineHeight": 25,
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      hello boy
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "marginBottom": 20,
            "textAlign": "center",
          },
        ]
      }
      uppercase={false}
    >
      An unexpected error came up
    </Text>
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "height": 20,
          },
        ]
      }
    />
  </View>
</View>
`;
